src/NuGetGallery/Security/SecurityPolicyService.cs

```
@@ -52,12 +86,108 @@ public SecurityPolicyResult Evaluate(SecurityPolicyAction action, HttpContextBas
         }
 
         /// <summary>
-        /// Create any supported policy handlers.
+        /// Check if a user is subscribed to one or more security policies.
         /// </summary>
-        private static IEnumerable<UserSecurityPolicyHandler> CreateUserPolicyHandlers()
+        public bool IsSubscribed(User user, IUserSecurityPolicySubscription subscription)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+            if (subscription == null)
+            {
+                throw new ArgumentNullException(nameof(subscription));
+            }
+
+            var subscribed = FindPolicies(user, subscription);
+            var required = subscription.Policies;
+
+            return required.All(rp => subscribed.Any(sp => sp.Equals(rp)));
+        }
+
+        /// <summary>
+        /// Subscribe a user to one or more security policies.
+        /// </summary>
+        public Task SubscribeAsync(User user, IUserSecurityPolicySubscription subscription)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+            if (subscription == null)
+            {
+                throw new ArgumentNullException(nameof(subscription));
+            }
+            
+            if (!IsSubscribed(user, subscription))
```

The confirmation prompt should help guard against duplicate clicks. I don't think a lock in the gallery (per instance) or optimistic concurrency in the DB (per row) is the solution here.

Now that we have the Subscription column, we could add a unique index (Name, Subscription)... assuming we wouldn't have a subscription contain multiple of the same policies differing in value only. We could silently ignore duplicate inserts (SqlException 2601). Thoughts?